firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -158,15 +162,17 @@ public InstallationResponse createFirebaseInstallation(
         writeFIDCreateRequestBodyToOutputStream(httpURLConnection, fid, appId);
 
         int httpResponseCode = httpURLConnection.getResponseCode();
+        requestLimiter.setNextRequestTime(httpResponseCode);
 
         if (httpResponseCode == 200) {
           return readCreateResponse(httpURLConnection);
         }
 
         logFisCommunicationError(httpURLConnection, appId, apiKey, projectID);
 
-        if (httpResponseCode == 429 || (httpResponseCode >= 500 && httpResponseCode < 600)) {
+        if (httpResponseCode >= 500 && httpResponseCode < 600) {
           retryCount++;
+          shouldServerErrorRetry = retryCount <= MAX_RETRIES;
```

Comment:
I'm not sure if I like this code!
There is several things wrong with this I think:
a) When I read the While loop condition above, it is unclear to me what `shouldServerErrorRetry` is and I have to search the code to find where the variable is set.
b) If avoidable (which is almost always the case) local variables should **not** change in value, because it makes code hard to read - this is actually a great example why. If you have to do it at all costs, you should mark the local variable as `@Var`.
c) I just realize it's not a local variable, but an instance variable. Why is that? That is even more confusing to me. Are you saying that other parallel processes can change the value of the variable and affect this while loop? (c.1) I doubt that this is the case. (c.2) And if it's the case, I doubt that it's a good idea!
d) I also liked using a variable named `retryCount` in the while loop condition better than a variable called `shouldServerErrorRetry`... With the `retryCount` it is immediately obvious, that we will retry sending the same request for some times... `shouldServerErrorRetry` does not really give that much information.

I recommend:
Leave the `retryCount <= MAX_RETRIES` in the while loop.
Even better, get rid of the while loop and replace it with a `for (int retryCount = 0; retryCount <= MAX_RETRIES; retryCount++)`.
That explains so much better what is happening here than a local variable that is changed somewhere else.

Reply:
Changed it . PTAL.